apps/lsp/src/service/toc.ts

```diff
@@ -140,7 +141,7 @@ export class TableOfContents {
     }
 
     // compute restricted ranges (ignore headings in these ranges)
-    const isWithinIgnoredRange = isWithinRange(tokens, token => isCallout(token) || isTheorem(token) || isProof(token));
+    const isWithinIgnoredRange = isWithinRange(tokens, isDiv);
```

Comment:
Let me make things fun (of the type 3 variety).

What should we do about conditional content divs? (to say nothing of arbitrary Lua filters...)

Reply:
The behavior of conditional content divs is that the div itself will never actually appear in the resulting document. But if the conditions are met, then the contents will be included in the document, and headers inside will "move one level up".

Reply:
@vezwork No, I don't think that's the best move. Let's merge this as is and talk about a redesign for later.